src/fw_fs.rs

```diff
@@ -290,6 +290,11 @@ impl<'a> FirmwareVolume<'a> {
     /// Contents of the FirmwareVolume will be cached in this instance.
     pub unsafe fn new_from_address(base_address: u64) -> Result<Self, efi::Status> {
         let fv_header = &*(base_address as *const fv::Header);
+        if fv_header.signature != 0x4856465f {

```

Comment:
Can we do something where the signature is more obvious? Like (or whatever) I didn't test it.

```suggestion
        if fv_header.signature != u32::from_be_bytes(*b"FVH_") {
```

Reply:
Updated as requested, in the new check and the existing check. Thanks!